src/ModelContextProtocol.SourceGenerators/README.md

```diff
@@ -0,0 +1,103 @@
+# ModelContextProtocol.SourceGenerators
+
+This project contains source generators for the Model Context Protocol C# SDK.
+
+## XmlToDescriptionGenerator
+
+This source generator automatically creates `Description` attributes from XML documentation comments for partial methods tagged with `[McpServerTool]`.
```

Comment:
I feel that this type of source generator should belong much lower down the stack. In the MEAI AI functions infrastructure if not STJ itself. The problem of course is getting such a source generator to obey activation patterns specific to MCP attributes, but there are msbuild tricks we could employ to configure this at the individual project level.

Reply:
> The problem of course is getting such a source generator to obey activation patterns specific to MCP attributes, but there are msbuild tricks we could employ to configure this at the individual project level.

 Like what? I'm unclear what you would have this do at the meai layer... Look for any method used in a delegate that flows to AIFunctionFactory.Create?

Reply:
@halter73 / @mikekistler / @jeffhandley, before I spend time on this to get it in, any opinions?

Reply:
This sounds way more complicated than I imagined. I think this is a "nice to have" and not a "must have". If the judgement of the engineering team is that the cost/complexity doesn't justify the benefit, I'm on board with that.